x-pack/solutions/observability/plugins/apm/public/plugin.ts

```diff
@@ -371,6 +385,38 @@ export class ApmPlugin implements Plugin<ApmPluginSetup, ApmPluginStart> {
       },
     });
 
+    plugins.discoverShared.features.registry.register({
+      id: 'observability-traces-fetch-root-span-by-trace-id',
+      fetchRootSpanByTraceId: async (params, signal) => {
+        const { fetchRootSpanByTraceId } = await getApmDataHelper();
+        return fetchRootSpanByTraceId(params, signal);
+      },
+    });
+
+    plugins.discoverShared.features.registry.register({
+      id: 'observability-traces-fetch-span',
+      fetchSpan: async (params, signal) => {
+        const { fetchSpan } = await getApmDataHelper();
+        return fetchSpan(params, signal);
+      },
+    });
+
+    plugins.discoverShared.features.registry.register({
+      id: 'observability-traces-fetch-latency-overall-transaction-distribution',
+      fetchLatencyOverallTransactionDistribution: async (params, signal) => {
+        const { fetchLatencyOverallTransactionDistribution } = await getApmDataHelper();
+        return fetchLatencyOverallTransactionDistribution(params, signal);
+      },
+    });
+
+    plugins.discoverShared.features.registry.register({
+      id: 'observability-traces-fetch-latency-overall-span-distribution',
+      fetchLatencyOverallSpanDistribution: async (params, signal) => {
+        const { fetchLatencyOverallSpanDistribution } = await getApmDataHelper();
+        return fetchLatencyOverallSpanDistribution(params, signal);
+      },
+    });
+
```

Comment:
Do you think we should move this registrations to a separate file to clean the plugin up?